app/javascript/app/document-capture/components/file-input.jsx

```diff
@@ -66,9 +138,17 @@ function FileInput({ label, hint, bannerText, accept, value, onChange, className
        * See: https://github.com/airbnb/javascript/pull/2136
        */}
       {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-      <label htmlFor={inputId} className="usa-label">
+      <label
+        htmlFor={inputId}
+        className={['usa-label', allErrors.length && 'usa-label--error'].filter(Boolean).join(' ')}
```

Comment:
Thinking it might be time to either create a utility for conditional classname generating, or pull in one of the many available popular options (e.g. [`classnames`](https://www.npmjs.com/package/classnames), [`clsx`](https://github.com/lukeed/clsx)).

Reply:
I think this is fine for now -- the classnames and clsx seem neat but aren't as self explanatory, honestly I feel like I've seen `.filter(Boolean)` a few times in these PRs and it hasn't caused me much concern.

If this got too much longer (like if there were 3 or 4 classes with logic for each), and  we were looking for library-free ways to do something like this, we could probably just do a little function like this....?

```js
function labelClasses() {
  var classes = ['usa-label'];
  if (allErrors.length) { classes.push('usa-label--error'); }
  return classes.join(' ');
}
```

But we're not there yet so anyways, again, I think this is fine.

Reply:
Works for me as well. I've not felt too troubled by it, but the pattern is recurring. I'd hoped that something like `classnames` might _improve_ the odds that it's readily understood at a glance, but abstractions being as they are, I can see it going the other way as well.

Another nicety of these libraries is to be able to express a map of class names, which is a bit more readable when there are many conditional classes.

For example:

```js
// Before
const classes = [
    'btn',
    isPrimary && 'btn-primary btn-wide',
    isSecondary && 'btn-secondary',
    isUnstyled && 'btn-link',
    className,
]
    .filter(Boolean)
    .join(' ');

// After
const classes = classnames( 'btn', className, {
    'btn-primary btn-wide': isPrimary,
    'btn-secondary': isSecondary,
    'btn-link': isUnstyled,
} );
```

I still don't think the "Before" is necessarily difficult to understand, though. And since the line of thinking was primarily motivated depending how easily future developers might understand this, your opinion can serve as testimony that it might be premature to go this route 😄